pet-shop/src/Components/Navbar/DropdownMenu.jsx

```diff
@@ -1,12 +1,44 @@
-export default function DropdownMenu({showDropDown}){
+export default function DropdownMenu({showDropDown, animals, setAnimals, orignalAnimals}){
+
+    const filterDogs = () => {
+            const dogsOnly = animals.filter((animal) => {
+            return animal.type === 'dog';            
+        })
+        setAnimals(dogsOnly);
+    }
+
```

Comment:
So I think the main problem is that you call `setAnimals()` here with just dogs, so this updates the main animals list. Late on when you filter to catsOnly - it is using the filtered animal list - which is dogsOnly at that point

Reply:
Looking through all of these functions - they are exactly the same aside from a single string (e.g.) "dog".

This means they should be refactored into a single function definition with a parameter given to them:

```
   const filterAnimals = (animalName) => {
        const filteredList = animals.filter((animal) => {
            return animal.type === animalName;
        })
        setAnimals(filteredList);
    }

```

and

```
<li onClick={() => filterAnimals('dog')}>Dogs</li>
```

Reply:
It also begs the question - should we define this function at the base level where the animals are fetched and pass it in - instead of `setAnimals` as a prop? That way it would save having to pass through the list of animals also